grovedb/src/lib.rs

```diff
@@ -461,11 +460,31 @@ impl GroveDb {
         self.subtrees = self.temp_subtrees.drain().collect();
 
         // TODO: root tree actually does support transactions, so this
-        // code can be reworked to account for that
+        //  code can be reworked to account for that
+
+        // Free transactional data
         self.temp_root_tree = MerkleTree::new();
+        self.temp_root_leaf_keys = HashMap::new();
+        self.temp_subtrees = HashMap::new();
 
         Ok(db_transaction
             .commit()
             .map_err(PrefixedRocksDbStorageError::RocksDbError)?)
     }
+
+    /// Rollbacks previously started db transaction. For more details on the
+    /// transaction usage, please check [`GroveDb::start_transaction`]
+    pub fn rollback_transaction(
+        &mut self,
+        db_transaction: &OptimisticTransactionDBTransaction,
+    ) -> Result<(), Error> {
+        // Cloning all the trees to maintain to rollback transactional changes
```

Comment:
These are not required either if transaction would be consumed on rollback

Also, it leaves db in readonly mode and there is no way to start new transaction

Reply:
Agreed to keep it for now